front/pages/api/w/[wId]/use/chats/[cId]/messages/[mId]/index.ts

```diff
@@ -0,0 +1,236 @@
+    "chunks",
+  ],
+};
+
+export const chatMessageSchema: JSONSchemaType<ChatMessageType> = {
+  type: "object",
+  properties: {
+    sId: { type: "string" },
+    role: { type: "string" },
+    message: { type: "string", nullable: true },
+    retrievals: {
+      type: "array",
+      items: chatRetrievedDocumentSchema,
+      nullable: true,
+    },
+    query: { type: "string", nullable: true },
+    feedback: { type: "string", nullable: true },
+  },
+  required: ["role"],
+};
+
+export type ChatMessageResponseBody = {
+  message: ChatMessageType;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ChatMessageResponseBody | ReturnedAPIErrorType>
+): Promise<void> {
+  const session = await getSession(req, res);
+  const user = await getUserFromSession(session);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  if (!auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message: "Only users of the current workspace can retrieve chats.",
+      },
+    });
+  }
+
+  if (!(typeof req.query.cId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+
+  if (!(typeof req.query.mId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `mId` (stringg) is required.",
+      },
+    });
+  }
+
+  const dbChatSession = await ChatSession.findOne({
+    where: {
+      workspaceId: owner.id,
+      sId: req.query.cId,
+    },
+  });
+
+  if (!dbChatSession) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "chat_session_not_found",
+        message:
+          "The chat session for the message you're trying to modify was not found.",
+      },
+    });
+  }
+  const chatSession = {
+    ...dbChatSession,
+    created: dbChatSession.createdAt.getTime(),
+  };
+
+  switch (req.method) {
+    case "POST": {
+      const pRes = parse_payload(chatMessageSchema, req.body);
+      if (pRes.isErr()) {
+        res.status(400).end();
+        return;
+      }
+      const m = pRes.value;
+      const message = await upsertChatMessage(
+        dbChatSession.id,
```

Comment:
As per last comment I think the ideal state is the following:
```
const chatSession = getChatSession(...)
[...]
upsertChatMessage(chatSession, m)
```

Don't we have the mId in the message itself here? We just want to check that they are the same but after it's cleaner to just use `m` ?